### DISCUSSION TITLE
如果接口返回错误 比如验证参数 失败 怎么让他不缓存？ 只有成功了才缓存

### DISCUSSION THREAD

**Discussion by Ya-hui** [alovajs/alova#574] [Q&A]
如题 比如有个查询接口列表， 后端可能会返回 {code: 20001, message: "缺少xxx参数", data: [],  success: "failed"}   像后端给的接口 肯定会有 失败的可能， 不可能说都100%成功， 我看了下文档 受控的缓存 好像是可以去 判断response 失败还是成功  成功都缓存， 但是像这种情况有便捷的解决办法吗？ 如果用受控的缓存 感觉 杀鸡焉用牛刀

**Comment by MeetinaXD** [comment]
参考文档：[全局的响应拦截器](https://alova.js.org/zh-CN/tutorial/getting-started/basic/global-interceptor#全局的响应拦截器)

响应时根据数据状态码抛出异常，则不会设置为缓存。

**Reply by Ya-hui** [reply]
这个跟我的需求不是特符合
我是开发的chrome扩展 
content_script 去通知 background 发起请求， background如果接口出错 或者报错的话 content_script 是捕获不到的。 他们不是运行在同一个域里。 我之前知道错误的消息机制是 content_script会拿到响应信息去做判断 。 如果在全局响应拦截器中 直接 new error 就会导致content_script也拿不到错误信息了。  我在包装一层？ 全局响应 new error 不让缓存， 然后 
```ts

let response = {}
try {
   response = await aliH5ApiAlova.Post('/putImage/1.0/', data, {
    params: {
      jsv: '2.7.4',
      type: 'originaljson'
    },
    cacheFor: {
      mode: 'restore',
      tag: browser.runtime.getManifest().version,
      expire: 60 * 30 * 1000
    }
  })
  return response

} catch(e) {
   return response;
}

```

您感觉这样可行吗？ 还会缓存吗？
